client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/index.tsx

```diff
@@ -172,6 +179,31 @@ const designSetup: Step = function DesignSetup( { navigation } ) {
 				selectedSiteCategory: categorization.selection,
 			};
 			submit?.( providedDependencies );
+		} else if ( isAnchorSite && _selectedDesign ) {
+			setPendingAction( async () => {
+				if ( ! currentUser ) {
+					return;
+				}
+
+				await createSite( {
+					username: currentUser?.username,
```

Comment:
You are checking for `currentUser` in line 184, I don't think it's necessary to use the `?.` operator here.